docs/Controllers.md

```diff
@@ -0,0 +1,120 @@
+# Controllers
+
+_(c) AMWA 2022, CC Attribution-NoDerivatives 4.0 International (CC BY-ND 4.0)_
+
+## Informative
+
+A Controller is Client software that interacts with the NMOS APIs to discover, connect and manage devices within a networked media system.
+
+* This document includes normative references to be followed when implementing a Controller.
+* This document covers how the Controller interacts with the NMOS APIs only.
+  It does not cover other features of the Controller software, such as presentation.
+* This document does not cover any requirements relating to where a Controller is additionally acting as a Node (e.g. receiving monitoring information via IS-07).
+
+## General
+
+### User
+
+Where this document refers to the "user" of a Controller, this includes both human operators who drive the Controller manually and automation systems that drive the Controller programmatically.
+
+### HTTP APIs
+
+#### Trailing Slashes
+
+Controllers appending paths to `href` type attributes MUST support URLs both with and without a trailing slash, to avoid doubled or missing slashes.
+
+Controllers performing requests other than `GET` or `HEAD` (i.e `PUT`, `POST`, `DELETE`, `OPTIONS` etc.) MUST use URLs with no trailing slash present.
+
+#### API Versions
+
+The versioning format is `v<MAJOR>.<MINOR>`
+* `MINOR` increments will be performed for non-breaking changes (such as the addition of attributes in a response)
+* `MAJOR` increments will be performed for breaking changes (such as the renaming of a resource or attribute)
+
+Versions MUST be represented as complete strings. Parsing MUST proceed as follows: separate into two strings, using the point (`.`) as a delimiter. Compare integer representations of `MAJOR`, `MINOR` version (such that v1.12 is greater than v1.5).
+
+Implementers of Controllers are RECOMMENDED to support multiple versions of the NMOS APIs simultaneously in order to ease the upgrade process in live facilities.
+
+#### API Common Keys
+
+Controllers SHOULD follow the requirements for common API keys specified in the [IS-04 APIs: Common Keys](APIs%20-%20Common%20Keys.md) document including the requirements regarding [use of URNs](APIs%20-%20Common%20Keys.md#use-of-urns).
```

Comment:
Actually this is another awkward instance of "layering" conformance language; writing SHOULD here effectively weakens all of the conformance language in the referenced document. I think this needs to be rewritten as a statement of fact, "The requirements for common API keys [...] apply to Controllers."

The `/transporttype` endpoint is a fixed enumeration at IS-05 v1.1 - ([transporttype-response-schema.json](https://github.com/AMWA-TV/is-05/blob/v1.1.x/APIs/schemas/transporttype-response-schema.json)) - so the extensibility implied by URNs is not yet used.

I'm not sure, therefore, whether there is actually a sensible test case right now, except that Controllers handle _all_ transport types.

```suggestion
The requirements in the [APIs: Common Keys](APIs%20-%20Common%20Keys.md) section of the IS-04 specification, including those regarding [handling of URNs](APIs%20-%20Common%20Keys.md#use-of-urns), apply generally to Controllers' interactions with the NMOS APIs.
```

Reply:
ARG comment: are we happy with the cross-link to IS-04?

Some options:

1. MUST (and fix link to point at IS-04 APIs: Common Keys)
2. MUST and duplicate IS-04 APIs: Common Keys: Use of URNs only into IS-05
3. Delete this section completely since it only applies to correct handling of `controls` in IS-04 (and in the future to the _/transporttye_ endpoint)
4. Move this section down to Interoperability: IS-04: Discovery section in this doc
5. ... 🤔 💭 ...

Reply:
Resolved